qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -393,16 +400,26 @@ def validate_parameters(self) -> None:
         if self.is_parameterized():
             return
 
-        args = (self.duration, self.amp, *self._param_vals)
-        if self._consts_expr is not None and not bool(self._callable_consts(*args)):
-            param_repr = ", ".join(f"{p}={v}" for p, v in self.parameters.items())
-            const_repr = str(self._consts_expr)
-            raise PulseError(
-                f"Assigned parameters {param_repr} violate following constraint: {const_repr}."
-            )
+        if any(p.imag != 0 for p in self._param_vals):
+            raise PulseError("Pulse parameters must be real numbers except for 'amp'.")
+
+        if self.constraints is not None:
+            func_args = []
+            params = self.parameters
+            for name in sorted(map(lambda s: s.name, self.constraints.free_symbols)):
+                func_args.append(params[name])
+
+            if not bool(self._constraints_lambdify(*func_args)):
+                param_repr = ", ".join(f"{p}={v}" for p, v in self.parameters.items())
+                const_repr = str(self.constraints)
+                raise PulseError(
+                    f"Assigned parameters {param_repr} violate following constraint: {const_repr}."
+                )
 
-        if self.limit_amplitude and _validate_amplitude_limit(self):
+        if self.limit_amplitude and (np.abs(self._amp) > 1.0 or _validate_amplitude_limit(self)):
```

Comment:
I was thinking of the similar stuff, and alternative approach would be adding `strict_eval: bool` option in the validate method. I think we can come back to this in the follow-up.

Reply:
Another idea to come back to might be `eval_conditions` which is another expression like `constraints` but defines conditions when `eval` needs to be used, with otherwise just `abs(amp) < 1` used. Then for DRAG the condition could be `abs(beta) > sigma`.